### PR TITLE
Gamnit on iOS

### PR DESCRIPTION
**Pull request by xymus**
Support iOS as a target for Gamnit games, including:

* GC on iOS and Xcode assets folder support in the compiler.
* iOS view controller and OpenGL view using the GLKit services.
* Loading of assets: sound, texture and text (and thus 3D models and fonts).
* iOS touch event and orientation change support.
* A few bug fix and tweaks to Gamnit.
* Minimal makefile rules to target iOS.

The game `asteronits` works as expected, here it is on iPad (With controls that are way too big):

screen shot 2018-01-14 at 10 11 17

Limitations:
* By default, the compiler generates an app for the simulator. To compile for a physical device, in practice, I run `nitc -m ios --compile-dir nit_compile` to generate the iOS project and view errors in Nit or in the custom Objective-C code. And then I launch Xcode with `open -a xcode nit_compile/ios/my_project.xcodeproj` to compile and run on a device from there.

* The depth API (3D) does not work on iOS with this PR because of known bugs in the current implementation. I have a rewrite of the depth API that fixes these bugs and bring a much needed performance boost.

* A `Sound` instance can't be played more than once concurrently. This could be improved by using `AVPlayer` as kind of sound channel and an `AVPlayerItem` for each sound instance, instead of the current `AVAudioPlayer`.

* iOS does not support exactly the same format for assets than Android. Sounds in mp3 format should work on both platforms. PNG files work on both platforms but iOS is more picky on the color palette, I've had an issue with a greyscale PNG, RGBA PNG files should be safe.

  You can always use the `app_files` annotation to include platform-specific assets.

* `gamnit::selection` does not work on iOS because we can't read the screen pixels. It should be updated to use an FBO.

* We still need a portable API for locking the screen orientation.

* Compiling Gamnit games for iOS fails at a missing pkg-config package. The package is not used and in can be safely ignored by removing the "pkgconfig" line in `lib/glesv2/glesv2.nit`. I'll need to find a clean fix for this...

### CODE REVIEW

**Comment by xymus** [comment]
test osx please

**Comment by xymus** [comment]
Added a few commit and tweaked the PR description to reflect the latest changes. The simulator now works as expected and I've added a note about the error on `pkgconfig`.

Thanks @jpcaissy for testing this PR!

File: lib/gamnit/input_ios.nit [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	redef fun touches_began(touches, event)
+	do app.accept_event(new GamnitIOSPointerEvent(event, touches.any_object, content_scale_factor, true, false))
+
+	redef fun touches_moved(touches, event)
+	do app.accept_event(new GamnitIOSPointerEvent(event, touches.any_object, content_scale_factor, true, true))
+
+	redef fun touches_ended(touches, event)
+	do app.accept_event(new GamnitIOSPointerEvent(event, touches.any_object, content_scale_factor, false, false))
+
+	# TODO handle cancel
```
Comment (BlackMinou) [inline lib/gamnit/input_ios.nit]: TODO + already some code ?

File: lib/gamnit/display_ios.nit [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+in "ObjC" `{
+	#import <GLKit/GLKit.h>
+	#import <OpenGLES/ES2/gl.h>
+`}
+
+redef class GamnitDisplay
+
+	redef var width = 200
+	redef var height = 300
```
[the remark this answers is not in the document]
Comment (xymus) [reply lib/gamnit/display_ios.nit]: This shouldn't happen but the app may access the display width and height before the call to `setup`. Adding checks would add complexity to the code, so I went the other way to set basic values that could work. It's not ideal...

File: lib/gamnit/gamnit_ios.nit [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+redef class App
+	redef fun did_finish_launching_with_options
+	do
+		create_gamnit
+		create_scene
+		return super
+	end
+
+	redef fun run do end
```
[the remark this answers is not in the document]
Comment (xymus) [reply lib/gamnit/gamnit_ios.nit]: The super `run`, in `gamnit.nit`, is the main game loop. This redef disables our custom loop to rely on the GLKView callbacks instead. I'll add some doc.

File: lib/gamnit/input_ios.nit (hunk 1 above)
Comment (xymus) [reply lib/gamnit/input_ios.nit]: The code may work, but I haven't tested it out yet. More experimentation is needed.

**Comment by xymus** [comment]
test osx please

**Comment by xymus** [comment]
Fixed support for spaces in the Nit directory, like on the test server.

test osx please

**Comment by xymus** [comment]
test osx please
test android please

**Comment by xymus** [comment]
test osx please

**Comment by xymus** [comment]
test osx please

**Pull request merged**